pkg/agent/clientset.go

```diff
@@ -151,6 +159,11 @@ func (cs *ClientSet) sync() {
 			duration = wait.Jitter(backoff.Duration, backoff.Jitter)
 		}
 		time.Sleep(duration)
+		select {
+			case <-cs.stopCh:
+				break
```

Comment:
I don't think this does what is intended. It'll break out of the select statement but the for loop will keep running. Probably need to add a flag?

Reply:
good catch

Reply:
Good catch. Fixed.